### PR TITLE
Add user personalization recipe

### PR DESCRIPTION
**Pull request by yuzhoujianxia**
*Issue #, if available:*

*Description of changes:*
Add new recipe user personalization and use exploration config the help customer explore cold items.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb [hunk 1]
```diff
@@ -0,0 +1,1271 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Steps\n",
+    "\n",
+    "This notebook shows how to use the new Amazon Personalize's user personalization recipe (aws-user-personalization) which balances recommendations between new and old items while allowing you to move the balance in favor of more new of old items.\n",
```
Comment (chwastm) [inline next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:9]: Please re-word to:

This notebook shows how to use Amazon Personalize's new user personalization recipe (aws-user-personalization).  This recipe balances recommendations between new and old items, allowing you to adjust the balance in favor of more new or more old items

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb [hunk 2]
```diff
@@ -0,0 +1,1271 @@
...
+    "\n",
+    "At a high level, using the new USER Personalization recipe involves the below steps:  \n",
+    "\n",
+    "1. Setup Personalize client\n",
+    "2. Create DatasetGroup, Schema, Datasets, ingest Interaction\n",
```
Comment (chwastm) [inline next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:14]: Create DatasetGroup, Define a Schema, Import Datasets and Ingest real-time Interactions

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb [hunk 3]
```diff
@@ -0,0 +1,1271 @@
...
+    "Let's first setup the client for personalize and s3."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# public s3 bucket in personalize service store the example dataset, you can change to your owns if you have.\n",
```
Comment (chwastm) [inline next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:98]: Can you re-word this.  Is this an Amazon Personalize owned bucket or should the user be adding their bucket here?  I'm not sure what this is saying

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb [hunk 4]
```diff
@@ -0,0 +1,1271 @@
...
+    "items_df.head(2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Items dataset\n",
+    "\n",
+    "Items dataset is similar to as before which contains Item_ids and associated metadata.\n",
```
Comment (chwastm) [inline next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:209]: what does simliar to as before mean?  before this feature was launched?

If that the case, just say the Items data contains Item_ids and metadata associate with those items

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb [hunk 5]
```diff
@@ -0,0 +1,1271 @@
...
+    "\n",
+    "**ITEM_ID**: Item Id  \n",
+    "\n",
+    "**genres** Metadata of the Item, if mutliple categorical then use pipe to concatenate.\n",
```
Comment (chwastm) [inline next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:213]: check spelling.  and say if multiple categorical items for the same record then use a pipe " | | to concatenate

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb [hunk 6]
```diff
@@ -0,0 +1,1271 @@
...
+    "\n",
+    "**creation_timestamp** Timestamp of when the item was added to catalog\n",
```
Comment (chwastm) [inline next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:215]: Timestamp when the item was added ...

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb [hunk 7]
```diff
@@ -0,0 +1,1271 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Update the timestamp\n",
+    "\n",
+    "This notebook shows how to use the new Amazon Personalize's user personalization recipe (aws-user-personalization) which balances recommendations between new and old items while allowing you to move the balance in favor of more new or old items.\n",
+    "\n",
+    "Since the timestamp of the demo dataset is not updated. Let's first update the timestamp of the interaction and item creation time close to 8 days ago. This can help the recommendation after put events more accurate."
```
Comment (chwastm) [inline next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:229]: Here we are going to update the stamp of our dataset to 8 days from today in order to show the impact of real-time interactions on our recommendations

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb [hunk 8]
```diff
@@ -0,0 +1,1271 @@
...
+    "items_df = items_df.astype({\"creation_timestamp\": 'int64'})\n",
+    "items_df.creation_timestamp = items_df.creation_timestamp + delta\n",
+    "items_df.to_csv(os.getcwd() + \"/items.csv\", index = False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### After update, let's check the internaction and items one more time"
```
Comment (chwastm) [inline next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:266]: spellcheck

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb [hunk 9]
```diff
@@ -0,0 +1,1271 @@
...
+    "        \n",
+    "    time.sleep(60)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4. Create a campaign\n",
+    "When create the campaign, we can set the itemExplorationConfig to configure cold-items exploration weight and also exploration age cut-off. For now, we can set higher explorationWeight as 0.9 and explorationItemAgeCutOff to 7, so we think all the item creation time less then 7 days would be considered as cold item, we would do more exploration on those new items."
```
Comment (chwastm) [inline next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:768]: when creating the campaign

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb (hunk 1 above)
Comment (yuzhoujianxia) [reply next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:9]: Changed

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb (hunk 2 above)
Comment (yuzhoujianxia) [reply next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:14]: Changed

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb (hunk 3 above)
Comment (yuzhoujianxia) [reply next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:98]: This is a personalize service owned s3 bucket, which used to store our demo dataset.

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb (hunk 6 above)
Comment (yuzhoujianxia) [reply next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:215]: Changed

File: next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb (hunk 5 above)
Comment (yuzhoujianxia) [reply next_steps/core_use_cases/user_personalization/user-personalization-with-exploration.ipynb:213]: Changed: Metadata of the Item, if multiple categorical classification for the same item then use a pipe " | " to concatenate.

**Comment by chrisking** [comment]
This notebook looks like it is for an entirely new algorithm/recipe which creates a new use case. Please look at the existing structure of use-cases and notebooks. Update the readmes accordingly to describe this, and then remove the other notebooks around metadata and cold starting that are not needed within that new use-cases' folder.

**Pull request merged**